File: cpp/p9.cpp
#include <iostream>

int findPythagoreanTriplet(int sum)
{
    for (size_t a = 1; a < sum / 3; a++)
    {
        for (size_t b = 0; b < sum / 2; b++)
        {
            int c = sum - a - b;
            if (a * a + b * b == c * c)
            {
                return a * b * c;
            }
        }
    }
}

int main(int argc, char const *argv[])
{
    int ans = findPythagoreanTriplet(1000);
    std::cout << ans << std::endl;
    return 0;
}

File: go/p9.go
package main

func findPythagoreanTriplet(sum int) (int, int, int) {
	for a := 1; a < sum/3; a++ {
		for b := a + 1; b < sum/2; b++ {
			c := sum - a - b
			if a*a+b*b == c*c {
				return a, b, c
			}
		}
	}
	return 0, 0, 0
}

func p9() int {
	sum := 1000
	a, b, c := findPythagoreanTriplet(sum)
    product := a * b * c
    return product
}
